src/dmd/identifier.d

```diff
@@ -138,6 +138,27 @@ nothrow:
         return idPool(buf.peekSlice());
     }
 
+    /***************************************
+     * Generate deterministic named identifier based on a source location,
+     * such that the name is consistent across multiple compilations.
+     * Params:
+     *      prefix      = first part of the identifier name, null-terminated.
+     *      loc         = source location to use is the identifier name.
+     * Returns:
+     *      Identifier (inside Identifier.idPool) with deterministic name based
+     *      on the source location.
+     */
+    static Identifier generateIdWithLoc(const(char)* prefix, const ref Loc loc)
```

Comment:
Any reason this can't be `string prefix` ?

Reply:
No, this should be `string`. Except when I do that:
`dmd/identifier.d(151): Error: string is used as a type`

Turns out `Identifier` has a `const char*` member called `string`!  >_<

Reply:
@WalterBright I've changed it to string, by prefixing with `.`. Ugly, but this PR shouldn't become a refactoring of `Identifier`. I'll do that now in another PR.

Reply:
The refactoring PR has been merged now, the leading dot can now be removed?

Reply:
@jacob-carlborg Yes, done.